src/Service/IncentiveService.php

```diff
@@ -0,0 +1,133 @@
+<?php
+
+namespace App\Service;
+
+use Doctrine\ORM\EntityManagerInterface;
+use App\Audit\Log;
+
+class IncentiveService
+{
+    protected $rdrApiService;
+    protected $siteService;
+    protected $userService;
+    protected $em;
+    protected $loggerService;
+
+    public function __construct(
+        RdrApiService $rdrApiService,
+        SiteService $siteService,
+        UserService $userService,
+        EntityManagerInterface $em,
+        LoggerService $loggerService
+    ) {
+        $this->rdrApiService = $rdrApiService;
+        $this->siteService = $siteService;
+        $this->userService = $userService;
+        $this->em = $em;
+        $this->loggerService = $loggerService;
+    }
+
+    public function getRdrObject($incentive, $type = 'create')
+    {
+        $obj = new \StdClass();
+        $email = $this->userService->getUser()->getEmail();
+        $now = new \DateTime();
+        if ($type !== 'create') {
+            $obj->incentiveId = $incentive->getRdrid();
+        }
+        if ($type === 'cancel') {
+            $obj->cancelledBy = $email;
+            $obj->cancelledDate = $now;
+            $obj->cancel = true;
+        } else {
+            $obj->createdBy = $email;
+            $obj->site = $this->siteService->getSiteIdWithPrefix();
+            $obj->dateGiven = $incentive->getIncentiveDateGiven();
+            $obj->occurrence = $incentive->getOtherIncentiveOccurrence() ?? $incentive->getIncentiveOccurrence();
+            $obj->incentiveType = $incentive->getOtherIncentiveType() ?: $incentive->getIncentiveType();
+            if ($incentive->getGiftCardType()) {
+                $obj->giftcardType = $incentive->getGiftCardType();
+            }
+            $obj->amount = $incentive->getIncentiveAmount();
+            $obj->notes = $incentive->getNotes();
+        }
+        return $obj;
+    }
+
+    public function createIncentive($participantId, $incentiveForm)
```

Comment:
It would be good to have tests for these other methods too. It will be a bit more complicated than the getRdrObject method, but I think it can be done by mocking the RDR, user, and site services.